akka-testkit/src/test/scala/akka/testkit/Coroner.scala

```diff
@@ -144,6 +144,11 @@ object Coroner {
                 #Heap usage: ${memMx.getHeapMemoryUsage()}
                 #Non-heap usage: ${memMx.getNonHeapMemoryUsage()}""".stripMargin('#'))
 
+    // If we look too soon, we've seen the JVM report a thread
+    // (in that case deadlock-thread-a from CoronerSpec)
+    // waiting on the lock while no thread seemed to be holding it
+    Thread.sleep(300)
```

Comment:
Oh wow. That's a hell of an extra-cost. 😞

Reply:
Hmm, this cost only will be paid in case the report is printed (which is very rare).